src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -610,11 +620,13 @@ private ProcessingBranch PrepareProcessingBranch(Block suggestedBlock, Processin
                 // we need to dig deeper to go all the way to the false (reorg boundary) head
                 // otherwise some nodes would be missing
                 bool notFoundTheBranchingPointYet = !_blockTree.IsMainChain(branchingPoint.Hash!);
-                bool notReachedTheReorgBoundary = branchingPoint.Number > (_blockTree.Head?.Header.Number ?? 0);
+                bool notReachedTheReorgBoundary = (options & ProcessingOptions.Trace) == ProcessingOptions.Trace
```

Comment:
Why does it need trace now?

Reply:
Marek suggested this to make sure we exit on first iteration even if client was synced with errors

Reply:
wait @OlegJakushkin check this comment:
https://github.com/NethermindEth/nethermind/pull/5146/files#r1073288034